ui/components/app/auto-detect-token/auto-detect-token-modal.tsx

```diff
@@ -0,0 +1,141 @@
+  const { chainId } = useSelector(getProviderConfig);
+
+  const handleTokenAutoDetection = useCallback(
+    (val) => {
+      trackEvent({
+        event: val
+          ? MetaMetricsEventName.TokenAutoDetectionEnableModal
+          : MetaMetricsEventName.TokenAutoDetectionDisableModal,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          chain_id: chainId,
+          locale: getCurrentLocale,
+          referrer: ORIGIN_METAMASK,
+        },
+      });
+      dispatch(setUseTokenDetection(val));
+      onClose(val);
+    },
+    [dispatch],
+  );
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => onClose(true)}
+      isClosedOnOutsideClick={false}
+      isClosedOnEscapeKey={false}
+      className="mm-modal__custom-scrollbar mm-auto-detect-in-modal"
+      autoFocus={false}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          alignItems={AlignItems.center}
+          justifyContent={JustifyContent.center}
+        >
+          Enable token autodetection
+        </ModalHeader>
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          paddingLeft={4}
+          paddingRight={4}
+        >
+          <Box
+            className="image-default"
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.center}
+            borderRadius={BorderRadius.SM}
+          />
+          <Text
+            variant={TextVariant.bodyMd}
+            textAlign={TextAlign.Justify}
+            padding={0}
+          >
+            {t('allowMetaMaskToDetectTokens')}
+            <Box textAlign={TextAlign.Justify} paddingLeft={2}>
+              <Text variant={TextVariant.inherit} as="li" paddingTop={2}>
+                {t('immediateAccessToYourTokens')}
+              </Text>
+              <Text variant={TextVariant.inherit} as="li">
+                {t('effortlesslyNavigateYourDigitalAssets')}
+              </Text>
+              <Text variant={TextVariant.inherit} as="li">
+                {t('diveStraightIntoUsingYourTokens')}
+              </Text>
+            </Box>
+          </Text>
+          <Box>
+            <Box paddingTop={8}>
+              <ButtonPrimary
+                onClick={() => handleTokenAutoDetection(true)}
+                variant={TextVariant.bodyMd}
+                size={ButtonPrimarySize.Md}
+                block
+              >
+                {t('allow')}
+              </ButtonPrimary>
+            </Box>
+            <Box paddingTop={4} paddingBottom={4}>
+              <ButtonSecondary
+                onClick={() => handleTokenAutoDetection(false)}
+                variant={TextVariant.bodyMd}
+                size={ButtonSecondarySize.Md}
+                block
+              >
+                {t('notRightNow')}
+              </ButtonSecondary>
+            </Box>
+          </Box>
+        </Box>
```

Comment:
You could use `ModalFooter` here either following DS button standards
```jsx
  <ModalFooter
    onSubmit={() => handleTokenAutoDetection(true)}
    submitButtonProps={{
      children: t('allow'),
      paddingLeft: 10, // this evens out buttons still trying to find a dynamic way to allow for this
      paddingRight: 10,
    }}
    onCancel={() => handleTokenAutoDetection(false)}
    cancelButtonProps={{
      children: t('notRightNow'),
    }}
  />
```

Or add `block` prop if you need full width
```jsx
<ModalFooter
  onSubmit={() => handleTokenAutoDetection(true)}
  submitButtonProps={{
    children: t('allow'),
    block: true,
  }}
  onCancel={() => handleTokenAutoDetection(false)}
  cancelButtonProps={{
    children: t('notRightNow'),
    block: true,
  }}
/>
```